app/Controllers/Http/EventsController.ts

```diff
@@ -32,21 +33,20 @@ export default class EventsController {
     const { cover } = await request.validate(CoverUploadValidator)
     event.cover = cover ? Attachment.fromFile(cover) : null
 
-    if (auth.user) {
-      // TODO set is_public if user.role equal moderator or admin
-      event.creatorEmail = auth.user.email
-      await event.save()
-    }
-
     await event.save()
     response.json({ success: true, event })
   }
 
   // update one event with id
-  public async update({ request, response }: HttpContextContract) {
+  public async update({ request, response, auth }: HttpContextContract) {
     const { id } = request.params()
     const event = await Event.findOrFail(id)
 
+    if (!(await EventService.isAllowedModifyOnwEvent(event, auth.user))) {
+      return response
+        .status(403)
+        .json({ success: false, message: 'Du hast nicht die Berechtigung dieses Event zu ändern!' })
+    }
```

Comment:
Das liest sich etwas verwirrend.

'wenn dir nicht erlaubt ist dein eigenes event zu bearbeiten, darfst du es nicht'
Ich würde die funktion umbenennen, Vielleicht in sowas wie 'createdThisEvent' oder so

Reply:
Außerdem fragst du immer nach der Negation. Warum nicht die funktion dann umdrehen? In 'isNotAllowedToModifyEvent'?

Reply:
und modifyOwn event ist ein ganz anderes Ticket. Ich bin etwas verwirrt